### PR TITLE
Use strpos instead of substr for maintainability

### PR DESCRIPTION
**Pull request by GaryJones**
`substr()` invokes additional memory allocation, which is not needed in the context. `strpos()` will do the same job, but without any overhead (it's just searching a string).

### CODE REVIEW

File: php/utils.php [hunk 1]
```diff
@@ -541,7 +541,7 @@ function parse_url( $url ) {
  * @return bool
  */
 function is_windows() {
-	return false !== ( $test_is_windows = getenv( 'WP_CLI_TEST_IS_WINDOWS' ) ) ? (bool) $test_is_windows : strtoupper( substr( PHP_OS, 0, 3 ) ) === 'WIN';
+	return false !== ( $test_is_windows = getenv( 'WP_CLI_TEST_IS_WINDOWS' ) ) ? (bool) $test_is_windows : 0 === stripos( PHP_OS, 'WIN' );
```
Comment (GaryJones) [inline php/utils.php]: Travis failing here, with complaint about it not following Yoda conditions.

**Comment by GaryJones** [comment]
I'm refactoring the code in `is_windows()` to hopefully avoid what I think is a false positive, so will try adding a unit test as well.

**Comment by GaryJones** [comment]
This Travis failure seems unrelated to the code that's been changed?

**Comment by danielbachhuber** [comment]
> This Travis failure seems unrelated to the code that's been changed?

Probably not:

image

Let's pass on this change for now

**Comment by GaryJones** [comment]
Ah - my fault for not interpreting the message correctly.

I've pushed a new [commit](https://github.com/GaryJones/wp-cli/commit/ac3f2eac878ed0f0e0e579aa41b62587ec0738be#diff-1847f885cd7cdf69f9fbad54754c04fdR297) to address the int vs string type [difference](https://github.com/wp-cli/wp-cli/pull/4417/files#diff-1847f885cd7cdf69f9fbad54754c04fdR297).

**Comment by GaryJones** [comment]
> I'm refactoring the code in is_windows() to hopefully avoid what I think is a false positive, so will try adding a unit test as well.

FWIW, the false positive has been [addressed](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/pull/1184).

I can understand why the PR has been closed, but I'll leave this [comment](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/1181#issuecomment-334726522) by @jrfnl here:

> "In the mean time, for the love of code, can someone refactor that line to a readable code block ?"

**Pull request closed without merging**